src/WeakReference.mo

```diff
@@ -0,0 +1,30 @@
+/// Module that implements a weak reference to an object.
+///
+/// ATTENTION: This functionality does not work with classical persistence (`--legacy-persistence` moc flag).
+///
+/// Usage example:
+/// Import from the core package to use this module.
+/// ```motoko name=import
+/// import WeakReference "mo:core/WeakReference";
+/// ```
+
+import Prim "mo:⛔"
+
+module {
+  type WeakReference<T> = {
+    ref : weak T
+  };
+
+  public func allocateWeakRef<T>(obj : T) : WeakReference<T> {
+    return { ref = Prim.allocWeakRef<T>(obj) }
+  };
+
+  public func deref<T>(weakRef : WeakReference<T>) : ?T {
```

Comment:
`WeakReference.get`? This would be more consistent with other languages such as Java and C#.